kick_fifo.asm

```diff
@@ -2,49 +2,75 @@
 
```

Comment:
It'd be good to take a detailed look at this change just to make sure I'm not doing something stupid